Zend/zend_execute.c

```diff
@@ -2012,6 +2012,85 @@ static zend_always_inline void zend_fetch_property_address(zval *result, zval *c
 	}
 }
 
+static zend_always_inline zval *zend_fetch_static_prop_address(zend_execute_data *execute_data, zval *varname, uint32_t varname_op_type, const znode_op *class, uint32_t class_op_type, void **polymorphic_cache_slot, void **ce_cache_slot)
+{
+	zval *retval;
+	zend_string *name;
+	zend_class_entry *ce;
+
+	if (varname_op_type == IS_CONST) {
+		name = Z_STR_P(varname);
+	} else if (EXPECTED(Z_TYPE_P(varname) == IS_STRING)) {
+		name = Z_STR_P(varname);
+		zend_string_addref(name);
+	} else {
+		if (varname_op_type == IS_CV && UNEXPECTED(Z_TYPE_P(varname) == IS_UNDEF)) {
+			zval_undefined_cv(EX(opline)->op1.var, execute_data);
+		}
+		name = zval_get_string(varname);
+	}
+
+	if (class_op_type == IS_CONST) {
+		if (polymorphic_cache_slot && EXPECTED((ce = CACHED_PTR_EX(polymorphic_cache_slot)) != NULL)) {
+			retval = CACHED_PTR_EX(polymorphic_cache_slot + 1);
+			/* check if static properties were destoyed */
+			if (UNEXPECTED(CE_STATIC_MEMBERS(ce) == NULL)) {
+				zend_throw_error(NULL, "Access to undeclared static property: %s::$%s", ZSTR_VAL(ce->name), ZSTR_VAL(name));
+				if (varname_op_type != IS_CONST) {
+					zend_string_release(name);
+				}
```

Comment:
Uh, does it actually _require_ it being const, or does it just happen that it's only used for when var_op_name is const? If it's the latter, that check anyway doesn't harm as it will be compiled out when inlining?